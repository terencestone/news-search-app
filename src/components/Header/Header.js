import React from 'react'
import PropTypes from 'prop-types'
import styles from './style.scss'
import CssModules from 'react-css-modules'
import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar'

export class HeaderComponent extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      buttonText: 'Outside button'
    };
  }

  changeText = () => {
    this.setState({ buttonText: 'Outer Button' })
  }

  render() {
    const { message } = this.props.testSaga
    console.log(message)

    return (
      <div>
        <Toolbar>
          <ToolbarGroup>
            <div>{`${process.env.REACT_APP_NAME}-v${process.env.REACT_APP_VERSION}`}</div>
            <button styleName='button'><span styleName='inside'>Vanity Button</span></button>
            <button onClick={this.changeText} styleName='inside'>{this.state.buttonText}</button>
          </ToolbarGroup>
        </Toolbar>
      </div>
    )
  }

  static propTypes = {
    testSaga: PropTypes.object
  }
}

export default CssModules(HeaderComponent, styles)
