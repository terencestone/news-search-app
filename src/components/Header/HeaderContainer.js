import { connect } from 'react-redux'

import HeaderComponent from './Header'

import { fetchArticles } from '../../redux/actions/actionCreators'

const mapStatetoProps = state => {
  return {
    // testSaga: state.testSaga
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchArticles: params => {
      dispatch(fetchArticles(params))
    }
  }
}

export default connect(mapStatetoProps, mapDispatchToProps)(HeaderComponent)
