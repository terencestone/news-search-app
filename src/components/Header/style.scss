.button{
  background-color: #202020;
  padding: 12px;
  border-radius: 2px;
  width: 100px;

  .inside {
    color: red;
  }
}

.inside {
  color: green;
}