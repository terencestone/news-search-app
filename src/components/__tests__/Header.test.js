import React from 'react';
import { HeaderComponent } from '../Header/Header';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import { mount, shallow } from "enzyme";

// Basic jest snapshot to catch any glaring bugs/changes to component
// MuiThemeProvider necessary in order for jest to render components properly with material-ui
test('renders correctly', () => {
  const component = renderer.create(
    <MuiThemeProvider>
      <HeaderComponent testSaga={{message: 'Hello'}} />
    </MuiThemeProvider>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// More granular enzyme tests
describe("HeaderComponent", () => {
  let props;
  let mountedHeaderComponent;
  const header = () => {
    if (!mountedHeaderComponent) {
      mountedHeaderComponent =
        shallow(<HeaderComponent {...props} />)
    }
    return mountedHeaderComponent;
  }

  beforeEach(() => {
    props = {
      testSaga: {message: 'Hello'},
    };
    mountedHeaderComponent = undefined;
  });

  describe("the rendered div", () => {
    it("contains everything else that gets rendered", () => {
      const divs = header().find("div");
      // When using .find, enzyme arranges the nodes in order such
      // that the outermost node is first in the list. So we can
      // use .first() to get the outermost div.
      const wrappingDiv = divs.first();
      // Enzyme omits the outermost node when using the .children()
      // method on header(). This is annoying, but we can use it
      // to verify that wrappingDiv contains everything else this
      // component renders.
      expect(wrappingDiv.children()).toEqual(header().children());
    });

    it("always renders two buttons", () => {
      expect(header().find("button").length).toBe(2);
    });

    it("renders initial button text", () => {
      expect(header().find("button").last().text()).toBe('Outside button');
    });

    it("changes the button text", () => {
      header().instance().changeText();
      expect(header().instance().state.buttonText).toBe("Outer Button")
    });
  });
});