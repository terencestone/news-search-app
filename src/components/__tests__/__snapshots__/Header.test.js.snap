// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "rgb(232, 232, 232)",
        "boxSizing": "border-box",
        "display": "flex",
        "height": 56,
        "justifyContent": "space-between",
        "muiPrepared": true,
        "padding": "0px 24px",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
          "marginLeft": undefined,
          "marginRight": undefined,
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div>
        react-boilerplate-v1.0.0
      </div>
      <button
        styleName="button"
      >
        <span
          styleName="inside"
        >
          Vanity Button
        </span>
      </button>
      <button
        onClick={[Function]}
        styleName="inside"
      >
        Outside button
      </button>
    </div>
  </div>
</div>
`;
